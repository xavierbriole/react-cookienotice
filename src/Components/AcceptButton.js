// @flow

import React from 'react'
import styled from 'styled-components'

const Button = styled.button`
  border-style: none;
  color: #000000;
  width: 100%;
  font-size: 16px;
  background: #ffffff;
  border-radius: 24px;
  padding: 12px 24px;
  box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2),
    0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0, 0, 0, 0.12);
  transition: all 0.5s ease 0s;
  cursor: pointer;

  &:hover {
    background: #ffffff;
    box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2),
      0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12);
  }

  &:active {
    box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2),
      0px 8px 10px 1px rgba(0, 0, 0, 0.14), 0px 3px 14px 2px rgba(0, 0, 0, 0.12);
  }

  &:focus {
    outline: 0;
  }

  &:not(:first-child) {
    margin-left: 8px;
  }

  @media (max-width: 768px) {
    &:not(:first-child) {
      margin-left: 0;
    }

    font-size: 12px;
    margin: 5px 0;
  }

  @media (prefers-color-scheme: dark) {
    background: #323232;
    color: #ffffff;

    &:hover {
      background: #3a3a3a;
    }
  }
`

type Props = {|
  label: string,
  onButtonClick: () => void,
|}

export default class AcceptButton extends React.Component<Props> {
  render() {
    const { label, onButtonClick } = this.props

    return (
      <Button className='accept-button' onClick={onButtonClick}>
        <span>{label}</span>
      </Button>
    )
  }
}
