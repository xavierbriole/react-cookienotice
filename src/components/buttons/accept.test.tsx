import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import AcceptButton from './accept'

describe('AcceptButton', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('should render', () => {
    const { container } = render(
      <AcceptButton
        acceptButtonLabel='acceptButtonLabel'
        handleAcceptButtonClick={() => {}}
      />,
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should display accept button label', () => {
    const { getByText } = render(
      <AcceptButton
        acceptButtonLabel='acceptButtonLabel'
        handleAcceptButtonClick={() => {}}
      />,
    )

    expect(getByText('acceptButtonLabel')).toBeInTheDocument()
  })

  it('should call handleAcceptButtonClick', () => {
    const handleAcceptButtonClick = jest.fn()

    const { getByText } = render(
      <AcceptButton
        acceptButtonLabel='acceptButtonLabel'
        handleAcceptButtonClick={handleAcceptButtonClick}
      />,
    )

    const acceptButton = getByText('acceptButtonLabel')

    acceptButton.click()

    expect(handleAcceptButtonClick).toHaveBeenCalled()
  })
})
