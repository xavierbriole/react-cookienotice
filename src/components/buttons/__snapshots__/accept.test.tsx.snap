// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcceptButton should render 1`] = `
<button
  class="react-cookienotice-accept-button"
>
  <span>
    acceptButtonLabel
  </span>
</button>
`;
